kubernetes_env/kube_env.py

```diff
@@ -425,14 +411,161 @@ def handle_filter(args):
+    samples_dir = f"{ISTIO_DIR}/samples"
+    bookinfo_dir = f"{samples_dir}/bookinfo"
+    apply_cmd = "kubectl apply -f"
+    book_cmd = f"{apply_cmd} {bookinfo_dir}"
+    cmd = f"{apply_cmd} {YAML_DIR}/bookinfo-services.yaml && "
+    cmd += f"{apply_cmd} {YAML_DIR}/bookinfo-apps.yaml && "
+    cmd += f"{book_cmd}/networking/bookinfo-gateway.yaml && "
+    cmd += f"{book_cmd}/networking/destination-rule-reviews.yaml && "
+    cmd += f"{apply_cmd} {YAML_DIR}/storage.yaml && "
+    cmd += f"{apply_cmd} {YAML_DIR}/istio-config.yaml && "
+    cmd += f"{apply_cmd} {YAML_DIR}/productpage-cluster.yaml "
+    result = util.exec_process(cmd)
+    application_wait()
+    return result
+
+
+def deploy_online_boutique(platform):
+    if check_kubernetes_status() != util.EXIT_SUCCESS:
+        log.error("Kubernetes is not set up."
+                  " Did you run the deployment script?")
+        sys.exit(util.EXIT_FAILURE)
+
+    apply_cmd = "kubectl apply -f"
+    cmd = f"{apply_cmd} {ONLINE_BOUTIQUE_DIR}/release && "
+    cmd += f"{apply_cmd} {YAML_DIR}/storage.yaml && "
+    cmd += f"{apply_cmd} {YAML_DIR}/istio-config.yaml && "
+    cmd += f"{apply_cmd} {YAML_DIR}/frontend-cluster.yaml "
+    result = util.exec_process(cmd)
+    application_wait()
+    delete_loadgen_cmd = "kubectl delete pods,services -l name=loadgenerator"
+    result = util.exec_process(cmd)
+    application_wait()
+    return result
+
+def deploy_hotel_reservation():
+    if check_kubernetes_status() != util.EXIT_SUCCESS:
+        log.error("Kubernetes is not set up."
+                  " Did you run the deployment script?")
+        sys.exit(util.EXIT_FAILURE)
+
+    apply_cmd = "kubectl apply -f"
+    cmd = f"{apply_cmd} {HOTEL_RESERVATION_DIR}/kubernetes && "
+    cmd += f"{apply_cmd} {YAML_DIR}/storage.yaml && "
+    cmd += f"{apply_cmd} {YAML_DIR}/istio-config.yaml && "
+    cmd += f"{apply_cmd} {YAML_DIR}/frontend-cluster.yaml "
+    result = util.exec_process(cmd)
+    application_wait()
+    return result
+
+def remove_application(application):
+    if application == "BK":
+        remove_bookinfo()
+    elif application == "OB":
+        remove_online_boutique()
+    elif applicaiton == "HR":
+        remove_hotel_reservation()
+
+def remove_bookinfo():
+    # remove bookinfo
+    samples_dir = f"{ISTIO_DIR}/samples"
+    bookinfo_dir = f"{samples_dir}/bookinfo"
+    cmd = f"{bookinfo_dir}/platform/kube/cleanup.sh &&"
+    cmd += f"kubectl delete -f {YAML_DIR}/storage.yaml && "
+    cmd += f"kubectl delete -f {YAML_DIR}/productpage-cluster.yaml "
+    result = util.exec_process(cmd)
+    return result
+
+def remove_online_boutique():
+    # remove online boutique
+    delete_cmd = "kubectl delete -f"
+    cmd = f"{delete_cmd} {ONLINE_BOUTIQUE_DIR}/release && "
+    cmd += f"{delete_cmd} {YAML_DIR}/storage.yaml && "
+    cmd += f"{delete_cmd} {YAML_DIR}/frontend-cluster.yaml "
+    result = util.exec_process(cmd)
+    return result
+
+def remove_hotel_reservation():
```

Comment:
There is a lot of overlap here. I think the config matrix may also be able to help there.

Reply:
Yeah, good call.  Config matrix should be implemented now.